src/java/org/apache/cassandra/cql3/restrictions/TermSlice.java

```diff
@@ -28,6 +28,12 @@
 
 final class TermSlice
 {
+    /**
```

Comment:
`TermSlice#isSupportedBy(ColumnMetadata, Index)` seems to reject `NEQ` expressions. Thus, if we have:
```
CREATE TABLE t (k int PRIMARY KEY, v int);
CREATE INDEX ON t(v) USING 'sai';
```
The query `SELECT * FROM t WHERE v!=0;` will require `ALLOW FILTERING`, whereas for example the query `SELECT * FROM t WHERE v>0;` will be accepted without AF. I think both queries should be accepted without AF, shouldn't they?

We should probably include the new operators in `AllowFilteringTest`.

Reply:
Indeed, both should be accepted.

Reply:
It seems that `SELECT * FROM t WHERE v!=0` still requires `ALLOW FILTERING` after the last changes. The new check on `AllowFilteringTest#testAllowFilteringOnFirstClusteringKeyColumn` makes the test fail.

Also, `AllowFilteringTest` has test methods for different schemas. We should probably add the `NEQ` check to all the test methods, not only in `testAllowFilteringOnFirstClusteringKeyColumn`.

Reply:
Indeed, both should be accepted. Fixed.